test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1391,6 +1395,9 @@ protected static boolean isXPackTemplate(String name) {
         if (name.startsWith(".transform-")) {
             return true;
         }
+        if (name.startsWith(".deprecation-")) {
```

Comment:
Referencing #66413 here, because all this logic in this class about protecting xpack related templates wouldn't be needed if these component and composable index templates are system owned.